pkg/piecestore/rpc/server/server.go

```diff
@@ -53,24 +57,87 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (err error)
 	return server.Run(ctx)
 }
 
+//DirSize returns the total size of the files in that directory
+func DirSize(path string) (int64, error) {
+	var size int64
+	_, err := os.Stat(path)
+	if err != nil {
+		return 0, errors.New("path doesn't exists")
+	}
+	adjSize := func(_ string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			size += info.Size()
+		}
+		return err
+	}
+	err = filepath.Walk(path, adjSize)
+
+	return size, err
+}
+
 // Server -- GRPC server meta data used in route calls
 type Server struct {
-	DataDir string
-	DB      *psdb.DB
-	pkey    crypto.PrivateKey
+	DataDir        string
+	DB             *psdb.DB
+	pkey           crypto.PrivateKey
+	totalAllocated int64
 }
 
 // Initialize -- initializes a server struct
 func Initialize(ctx context.Context, config Config, pkey crypto.PrivateKey) (*Server, error) {
 	dbPath := filepath.Join(config.Path, "piecestore.db")
 	dataDir := filepath.Join(config.Path, "piece-store-data")
 
+	// read the allocated disk space from the config file
+	allocatedDiskSpace := config.AllocatedDiskSpace
+
+	// get the disk space details
+	// The returned path ends in a slash only if it represents a root directory, such as "/" on Unix or `C:\` on Windows.
+	rootPath := filepath.Clean(config.Path)
+	diskSpace, err := disk.Usage(rootPath)
+	if err != nil {
+		return nil, err
+	}
+	freeDiskSpace := int64(diskSpace.Free)
+
 	db, err := psdb.Open(ctx, dataDir, dbPath)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Server{DataDir: dataDir, DB: db, pkey: pkey}, nil
+	// get how much is currently used, if for the first time totalUsed = 0
+	totalUsed, err := db.SumTTLSizes()
+	if err != nil {
+		//first time setup
+		totalUsed = 0x00
+	}
+
+	// check your hard drive is big enough
+	// first time setup as a piece node server
+	if (totalUsed == 0x00) && (freeDiskSpace < allocatedDiskSpace) {
+		allocatedDiskSpace = freeDiskSpace
+		zap.S().Warnf("Disk space is less than requested allocated space, allocating = %d Bytes", allocatedDiskSpace)
+		return &Server{DataDir: dataDir, DB: db, pkey: pkey, totalAllocated: allocatedDiskSpace}, nil
+	}
+
+	// on restarting the Piece node server, assuming already been working as a node
+	// used above the alloacated space, user changed the allocation space setting
+	// before restarting
+	if totalUsed >= allocatedDiskSpace {
+		allocatedDiskSpace = totalUsed
```

Comment:
We don't want to increase the allocatedDiskSpace. We shouldn't change it in this case.

Reply:
done